x-pack/plugins/observability_onboarding/server/routes/flow/route.ts

```diff
@@ -148,16 +148,13 @@ const getProgressRoute = createObservabilityOnboardingServerRoute({
     const esClient =
       coreStart.elasticsearch.client.asScoped(request).asCurrentUser;
 
-    const dataset = savedObservabilityOnboardingState.state
-      ?.datasetName as string;
-    const namespace = savedObservabilityOnboardingState.state
-      ?.namespace as string;
+    const type = savedObservabilityOnboardingState.type;
 
     if (progress['ea-status']?.status === 'complete') {
       try {
         const hasLogs = await getHasLogs({
-          dataset,
-          namespace,
+          type,
+          state: savedObservabilityOnboardingState.state,
```

Comment:
do you think we need to pass in the `type` separately since it is already in the saved object?

Reply:
I prefer to pass the `type` explicitly